ugc/storage.cpp

```diff
@@ -1,31 +1,365 @@
 #include "ugc/storage.hpp"
+#include "ugc/serdes.hpp"
+#include "ugc/serdes_json.hpp"
 
 #include "indexer/feature_decl.hpp"
+#include "indexer/index.hpp"
+
+#include "platform/platform.hpp"
+
+#include "coding/file_name_utils.hpp"
+#include "coding/file_reader.hpp"
+#include "coding/file_writer.hpp"
+#include "coding/internal/file_data.hpp"
+
+#include <algorithm>
+#include <utility>
+
+#include "3party/jansson/myjansson.hpp"
 
 namespace ugc
 {
-Storage::Storage(std::string const & filename)
-  : m_filename(filename)
+using namespace std;
+
+namespace
+{
+string const kIndexFileName = "index.json";
+string const kUGCUpdateFileName = "ugc.update.bin";
+string const kTmpFileExtension = ".tmp";
+
+using Sink = MemWriter<vector<char>>;
+
+string GetUGCFilePath() { return my::JoinPath(GetPlatform().WritableDir(), kUGCUpdateFileName); }
+
+string GetIndexFilePath() { return my::JoinPath(GetPlatform().WritableDir(), kIndexFileName); }
+
+bool GetUGCFileSize(uint64_t & size)
+{
+  return GetPlatform().GetFileSizeByName(kUGCUpdateFileName, size);
+}
+
+void DeserializeUGCIndex(string const & jsonData, vector<Storage::UGCIndex> & res)
+{
+  if (jsonData.empty())
+    return;
+
+  DeserializerJsonV0 des(jsonData);
+  des(res);
+}
+
+string SerializeUGCIndex(vector<Storage::UGCIndex> const & indexes)
 {
-  Load();
+  if (indexes.empty())
+    return string();
+
+  auto array = my::NewJSONArray();
+  for (auto const & index : indexes)
+  {
+    vector<char> data;
+    {
+      Sink sink(data);
+      SerializerJson<Sink> ser(sink);
+      ser(index);
+    }
+
+    data.push_back('\0');
```

Comment:
@VladiMihaylenko , By the way, why not to use string instead of vector<char> here? This will help us to get rid of this boring explicit zero byte at the end.